application/src/main/java/org/thingsboard/server/service/security/system/DefaultSystemSecurityService.java

```diff
@@ -131,9 +132,18 @@ public SecuritySettings saveSecuritySettings(TenantId tenantId, SecuritySettings
 
     @Override
     public void validateUserCredentials(TenantId tenantId, UserCredentials userCredentials, String username, String password) throws AuthenticationException {
+        SecuritySettings securitySettings = self.getSecuritySettings(tenantId);
+        UserPasswordPolicy passwordPolicy = securitySettings.getPasswordPolicy();
+
+        if (passwordPolicy.getForceUserToResetPasswordIfNotValid()) {
+            try {
+                validatePasswordByPolicy(password, passwordPolicy);
+            } catch (DataValidationException e) {
+                throw new BadCredentialsException("Password does not pass validation. Please try again or reset password to valid one.");
```

Comment:
Let's change the error message to something like:
`The entered password violates our policies. If this is your real password, please reset it.`

@ashvayka Any suggestions?